main.tf

```
@@ -3,7 +3,7 @@ locals {
   certificate_backends       = ["ACM", "SSM"]
   mutual_enabled             = var.authentication_type == "certificate-authentication"
   federated_enabled          = var.authentication_type == "federated-authentication"
-  saml_provider_arn          = local.federated_enabled ? try(join("", aws_iam_saml_provider.default.*.arn), var.saml_provider_arn) : null
+  saml_provider_arn          = local.federated_enabled ? try(aws_iam_saml_provider.default[0].arn, var.saml_provider_arn) : null
```

Why this change ?

---

Instead of using a `try` here, can we use a `coalesce` instead ?

```suggestion
  saml_provider_arn          = local.federated_enabled ? coalesce(join("", aws_iam_saml_provider.default.*.arn), var.saml_provider_arn)
```

or would that return the same error ?